src/service/fixed-layer.js

```diff
@@ -241,74 +241,66 @@ export class FixedLayer {
    * @return {!Promise}
    */
   update() {
-    if (this.elements_.length == 0) {
-      return Promise.resolve();
-    }
-
     // Some of the elements may no longer be in DOM.
     /** @type {!Array<!ElementDef>} */
     const toRemove = this.elements_.filter(
         fe => !this.ampdoc.contains(fe.element));
     toRemove.forEach(fe => this.removeElement_(fe.element));
 
+    if (this.elements_.length == 0) {
+      return Promise.resolve();
+    }
+
     // Next, the positioning-related properties will be measured. If a
     // potentially fixed/sticky element turns out to be actually fixed/sticky,
     // it will be decorated and possibly move to a separate layer.
     let hasTransferables = false;
     return this.vsync_.runPromise({
       measure: state => {
-        const autoTops = [];
         const elements = this.elements_;
-        const styles = [];
+        const autoTops = [];
+        const win = this.ampdoc.win;
 
         // Notice that this code intentionally breaks vsync contract.
         // Unfortunately, there's no way to reliably test whether or not
         // `top` has been set to a non-auto value on all platforms. To work
         // this around, this code compares `offsetTop` values with and without
         // `style.top = auto`.
-
-        // 1. Set all style top to `auto` and calculate the auto-offset.
         for (let i = 0; i < elements.length; i++) {
-          const el = elements[i].element;
-          const style = computedStyle(this.ampdoc.win, el);
-          styles.push(style);
-          if (endsWith(style.position || '', 'sticky')) {
-            setStyle(el, 'position', 'fixed');
-          } else {
-            setStyle(el, 'top', 'auto');
-          }
+          const element = elements[i].element;
```

Comment:
Can you up to document all steps that take places in bulks. E.g. "Change bottom -> remeasure top -> revert bottom"  and explain the result